docs/source/scancode-toolkit/cli-reference/core-options.rst

```diff
@@ -0,0 +1,282 @@
+`Core` Options
+==============
+
+.. include::  /scancode-toolkit/rst_snippets/basic_options.rst
+
+.. _cli_core:
+
+.. include::  /scancode-toolkit/rst_snippets/core_options.rst
+
+``--generated`` Options
+-----------------------
+
+The ``--generated`` option classifies automatically generated code files with a flag.
+
+An example of using ``--generated`` in a scan::
+
+    ./scancode -clpieu --json-pp sample_generated.json samples --generated
+
+In the results, for each file the following attribute is added with it's corresponding True/False
+value ::
+
+    "is_generated": true
+
+For the samples folder, the following files have a true value for their is_generated attribute::
+
+    "samples/zlib/dotzlib/LICENSE_1_0.txt"
+    "samples/JGroups/licenses/apache-2.0.txt"
+
+..
+    [ToDo] Research and Write Better
+
+``--max-email`` Options
+-----------------------
+
+If in the files that are scanned, in individual files, there are a lot of emails (i.e lists) which
+are unneccesary and clutter the scan results, ``--max-email`` option can be used to report emails
+only up to a limit in individual files.
+
+Some important INTEGER values of the ``--max-email INTEGER`` option:
+
+- 0  - No limit, inlcude all emails.
+- 50 - Default.
+
+An example usage::
+
+    ./scancode -clieu --json-pp sample.json samples --max-email 5
+
+This only reports 5 email adresses per file and ignores the rest.
+
+``--max-url`` Options
+---------------------
+
+If in the files that are scanned, in individual files, there are a lot of links to other websites
+(i.e url lists) which are unneccesary and clutter the scan results, ``--max-url`` option can be
+used to report urls only up to a limit in individual files.
+
+Some important INTEGER values of the ``--max-url INTEGER`` option:
+
+- 0  - No limit, inlcude all urls.
+- 50 - Default.
+
+An example usage::
+
+    ./scancode -clieu --json-pp sample.json samples --max-url 10
+
+This only reports 10 urls per file and ignores the rest.
+
+``--license-score`` Options
+---------------------------
+
+..
+    [ToDo] Research and Write License Matching Better
+
+License matching strictness, i.e. how closely matched licenses are detected in a scan, can be
+modified by using this ``--license-score`` option.
+
+Some important INTEGER values of the ``--license-score INTEGER`` option:
+
+- **0**     - Default and Lowest Value, All matches are reported.
+- **100**    - Highest Value, Only licences with a much better match are reported
```

Comment:
We probably need a  section on scoring elements for license matches.

Reply:
Yeah, I too feel this needs elaboration, and so I've left a ToDo here. But I'd need your/other developers guidance on that. That being said, I still have some unpushed work, and some improvements to make. After that, I was thinking we should mark sections for improvement, and then you/others could guide me/discuss together what can be done.

Reply:
Yes, this can be added later and expanded by other developers/users. We can track this with a ticket for now: https://github.com/nexB/aboutcode/issues/28

Reply:
@majurg There are some more issues I need help on. They all are marked as [ToDo] in the .rst files in their respective locations. Should I open tickets for them?